### PR TITLE
Make it possible to disable the BackgroundPreinitializer

### PR DESCRIPTION
**Pull request by ayudovin**
this [enhancement](https://github.com/spring-projects/spring-boot/issues/13524)

### CODE REVIEW

**Comment by philwebb** [comment]
Thanks for the PR. I think we can simplify this by bypassing all initialization if the property is set. The initialization code doesn't actually need to run at all. The reason we currently do it is to kick off in the background so that something called later runs quickly.

Probably just returning early from [this method](https://github.com/spring-projects/spring-boot/blob/2af8a9617b4a67b41fd1fcdfc98a86548af01596/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/BackgroundPreinitializer.java#L58) if the property is set will be enough.

File: spring-boot-project/spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc [hunk 1]
```diff
@@ -69,6 +69,7 @@ content into your application. Rather, pick only the properties that you need.
 
 	# SPRING CORE
 	spring.beaninfo.ignore=true # Whether to skip search of BeanInfo classes.
+    spring.backgroundpreinitializer.ignore=true # Whether background preinitializer should be ignored.
```
Comment (snicoll) [inline spring-boot-project/spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc]: We shouldn't add this here as it's not managed the same way as a regular property. Let's document the property and its usage on the class Javadoc instead.

**Comment by snicoll** [comment]
Thanks @ayudovin, this is now merged in `master` with a [polish commit](https://github.com/spring-projects/spring-boot/commit/304c6a02ba9a526f9a2bbbd944ba47a4003f874c).

**Comment by ayudovin** [comment]
@snicoll, Thank you!

**Comment by gdespres** [comment]
Hi, 

Boolean.getBoolean(IGNORE_BACKGROUNDPREINITIALIZER_PROPERTY_NAME) is always false.
The call to the method System.getProperty is missing. You can confirm it ?

**Comment by dsyer** [comment]
Have a look at the java docs for `Boolean.getBoolean()`.

**Comment by gdespres** [comment]
Thanks @dsyer, it's a misunderstanding of this method. 

I didn't known this feature.

**Pull request closed without merging**